src/com/microchip/mplab/nbide/embedded/arduino/wizard/ProjectSetupStep.java

```diff
@@ -311,6 +308,15 @@ public void readSettings(WizardDescriptor settings) {
             view.platformLocationField.setText( currentPlatform.getRootPath().toString() );
         }
         
+        if( currentPlatform==null){
+            resolvePlatformFromPath();
```

Comment:
I'm not sure whether I understand the reason for this change. If the "currentPlatform" isn't non-null at this point, we are risking a NPE a few lines above (308). We could however move this logic to the line right after line 306 where we set the text for `view.platformLocationField`.